batavia/core/exceptions.js

```diff
@@ -22,16 +22,61 @@ BaseException.prototype.toString = function() {
 }
 
 BaseException.prototype.__str__ = function() {
-    if (this.msg) {
-        return this.msg
+    if (this.args !== undefined) {
+        if (this.args.length == 1) {
+            return this.args[0].toString()
+        }
+        // Multiple args. Format like a tuple.
+        let result = '(' + arg.toString()
+        for (arg in this.args.slice(1)) {
+            result += ', ' + arg.toString()
+        }
+        return result = ')'
     } else {
         return ''
     }
 }
 
 BaseException.prototype.__repr__ = function() {
-    if (this.msg) {
-        return this.name + '(' + this.msg + ',)'
+    if (this.args != undefined) {
+        let output = ''
+        let count = 0
+        
+        const parse = function(msg) {
+            if (typeof msg === "string") {
+                // Message is probably a Python string.
+                // Default behaviour: wrap in single quotes
+                // String has single quotes: wrap in double quotes
+                // String has both: wrap in single quotes & escape internal single quotes.
+                let wrap = "'"
+                if (msg.includes("'")) {
+                    if (msg.includes('"')) { // example: this.msg = '\'"'
+                        msg = msg.replace("'", "\\'")
+                    } else {
+                        wrap = '"'
+                    }
+                }
+                return wrap + msg + wrap
+            } else {
+                // Avoid circular import by calling repr directly.
```

Comment:
You can require builtins (and types) inside in the function and not get a circular import. It would only be circular if you require it at the top-level.

If you look at the code in the repr builtin, it is checking for a \_\_call__ property on the \_\_repr__ and using that if it exists before calling the flat \_\_repr__ property.

Usually using callables.call_method would be good here, but repr actually skips some of the call_method code where it uses getattr and getattribute. (https://docs.python.org/3/reference/datamodel.html#special-lookup).

Reply:
I thought that was the case, but it seemed like the file that defines the exceptions shouldn't share code with files that throw the exceptions. On the very first run, if something crashes, the error handler for it doesn't crash as well and obscure the cause?

Reply:
I see your point. I can't say for certain that there wouldn't be an issue there. I am still fleshing out my understanding of all the parts of the system here. 

@freakboy3742 Can you give us some advice here? Is it safe to use builtins inside the `__repr__` method of the exceptions? Should we extract the "call a `__repr__` behaviour" from the builtin to somewhere in core without any dependencies?

Reply:
It's okay to leave this as it is, so I'll resolve this.